test/react-native/features/device.feature

```diff
@@ -0,0 +1,100 @@
+Feature: Device data
+
+Scenario: Handled JS error
+  When I run "DeviceJsHandledScenario"
+  Then I wait to receive a request
+  And the exception "errorClass" equals "Error"
+  And the exception "message" equals "DeviceJsHandledScenario"
+  And the event "unhandled" is false
+
+  # Android
+  And the event "device.id" is not null
+  And the event "device.jailbroken" is false
+  And the event "device.locale" equals "en_US"
+  And the event "device.manufacturer" is not null
+  And the event "device.model" is not null
+  And the event "device.osName" equals "android"
+  And the event "device.osVersion" matches "^\d+$"
+  And the event "device.runtimeVersions.androidApiLevel" is not null
+  And the event "device.runtimeVersions.reactNative" matches "^\d+\.\d+\.\d+$"
+  And the event "device.runtimeVersions.reactNativeJsEngine" matches "jsc"
+  And the event "device.totalMemory" is not null
+  And the event "device.freeDisk" is not null
+  And the event "device.freeMemory" is not null
+  And the event "device.orientation" equals "portrait"
+  And the event "device.time" is a timestamp
+
+Scenario: Unhandled JS error
+  When I run "DeviceJsUnhandledScenario" and relaunch the app
+  And I configure Bugsnag for "DeviceJsUnhandledScenario"
+  Then I wait to receive a request
+  And the exception "errorClass" equals "Error"
+  And the exception "message" equals "DeviceJsUnhandledScenario"
+  And the event "unhandled" is true
+
+  # Android
+  And the event "device.id" is not null
+  And the event "device.jailbroken" is false
+  And the event "device.locale" equals "en_US"
+  And the event "device.manufacturer" is not null
+  And the event "device.model" is not null
+  And the event "device.osName" equals "android"
```

Comment:
How's this and other lines going to integrate with the iOS work?  Do we have the tools in place now to handle it before merging, even if we don't add the iOS values just yet?

Reply:
as I understand it, @Cawllec is going to add the mechanism for this so can be merged as-is…

Reply:
Yes, I'll handle this